forecasting/src/autogluon/forecasting/task/forecasting/dataset.py

```diff
@@ -0,0 +1,195 @@
+                  if index_column is None, we will assume that the dataset contains only one time series
+
+    time_column: time of a data, in the form "YYYY-MM-DD HH:MM:SS", we are assuming that each time series contains the same time sequence,
+                 and the freq in each time series does not change.
+
+    target_column: values used for prediction, integers.
+
+    output:
+    a new dataframe in the form that each line contains a time series
+    transformed example would be:
+    >>> X
+          index_column  2020-01-22  2020-01-23  2020-01-24
+    0            A           1           2           3
+    1            C           1           2           3
+    2            B           1           2           3
+
+    """
+    if index_column is None:
+        X = X[[time_column, target_column]]
+        X["index_column"] = ["time_series" for i in range(X.shape[0])]
+    time_list = sorted(list(set(X[time_column])))
+    freq = pd.infer_freq(time_list)
+    if freq is None:
+        raise ValueError("Freq cannot be inferred. Check your dataset.")
+
+    def reshape_dataframe(df):
+        """
+        for each time occurs in the dataset, we select the target value corresponding to
+        each time series, and use dataframe.pivot() to convert it to one column, where the column name is the
+        time, each row is the corresponding target value for each time series.
+        """
+        data_dic = {index_column: list(set(df[index_column]))}
+
+        for time in time_list:
+            tmp = df[df[time_column] == time][[index_column, time_column, target_column]]
+            tmp = tmp.pivot(index=index_column, columns=time_column, values=target_column)
+            tmp_values = tmp[time].values
+            data_dic[time] = tmp_values
+        return pd.DataFrame(data_dic)
+
+    X = reshape_dataframe(X)
+    return X, freq
+
+
+def transform_tabular_to_gluonts_dataset(X, freq, index_column=None):
+    """
+    transform a dataframe in the following form to a gluon-ts dataset
+    >>> X
+          index_column  2020-01-22  2020-01-23  2020-01-24
+    0            A           1           2           3
+    1            C           1           2           3
+    2            B           1           2           3
+
+    """
+    if index_column is not None:
+        target = X.drop(index_column, axis=1)
+    else:
+        target = X.copy()
+    target_values = target.values
+    date_list = target.columns
+    processed_X = ListDataset([
+        {
+            FieldName.TARGET: target,
+            FieldName.START: pd.Timestamp(date_list[0], freq=freq),
+        }
+        for (target,) in zip(target_values, )
+    ], freq=freq)
+    return processed_X, freq
+
+
+class TimeSeriesDataset:
+
+    def __init__(self, dataset_name=None, train_path=None, test_path=None, is_train=True, prediction_length=None,
+                 index_column="index", time_column="date", target_column="target"):
```

Comment:
should we allow user to specify `freq` as an optional argument (default value = None), or does this not make sense?

Reply:
I think the freq can always be inferred from the train_data?